src/actions/triggerActions.js

```diff
@@ -109,12 +110,12 @@ export function processTriggerSample(currentValue, device, samplingData) {
             triggerWindowOffset,
             device.capabilities.prePostTriggering
         );
-
         const from = indexToTimestamp(triggerStartIndex - shiftedIndex);
         const to = indexToTimestamp(currentIndex - shiftedIndex);
         dispatch(chartWindowUnLockAction());
         dispatch(chartWindowAction(from, to, to - from));
         dispatch(chartWindowLockAction());
+        dispatch(setTriggerOriginAction(triggerStartIndex));
         dispatch(setTriggerStartAction(null));
```

Comment:
Minor: if trigger start and origin are always set at the same place, I'd merge them into one action.

Reply:
setTriggerStartAction is also set when the trigger level is initially hit, so not always at the same place. We could merge them, but I'm not sure it would be cleaner.